signac/common/config.py

```diff
@@ -15,14 +15,47 @@
 PROJECT_CONFIG_FN = os.path.join(".signac", "config")
 USER_CONFIG_FN = os.path.expanduser(os.path.join("~", ".signacrc"))
 
-# TODO: Consider making this entire module internal and removing all its
-# functions from the public API.
-
 
 def _get_project_config_fn(path):
     return os.path.abspath(os.path.join(path, PROJECT_CONFIG_FN))
 
 
+def _raise_if_older_schema(root):
+    """Raise if an older schema version is detected at the search path.
+
+    Parameters
+    ----------
+    root : str
+        Directory to check schema for.
+
+    Raises
+    ------
+    IncompatibleSchemaVersion
+        If the project uses an older schema version that requires migration.
+    """
+    from ..contrib.errors import IncompatibleSchemaVersion
+    from ..contrib.migration import _get_config_schema_version
+    from ..version import SCHEMA_VERSION, __version__
+
+    schema_version = int(SCHEMA_VERSION)
+
+    try:
+        schema_version = _get_config_schema_version(root, schema_version)
+        assert schema_version != int(SCHEMA_VERSION), (
```

Comment:
Maybe an `if` is better here? This is one of those nit picky things but since `assert` isn't ran with `-O`

Reply:
Well, I sort of did that intentionally since this indicates developer error somewhere. This is really intended to be something more like a debug print. I can remove it or change to an if statement if you'd prefer though.